jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestListTemplate.java

```diff
@@ -0,0 +1,171 @@
+ *         "os": "linux",
+ *       }
+ *     },
+ *     {
+ *       "mediaType": "application/vnd.docker.image.manifest.v2+json",
+ *       "size": 7682,
+ *       "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
+ *       "platform": {
+ *         "architecture": "amd64",
+ *         "os": "linux",
+ *         "features": [
+ *           "sse4"
+ *         ]
+ *       }
+ *     }
+ *   ]
+ * }
+ * }</pre>
+ *
+ * @see <a href="https://docs.docker.com/registry/spec/manifest-v2-2/#manifest-list">Image Manifest
+ *     Version 2, Schema 2: Manifest List</a>
+ */
+public class V22ManifestListTemplate implements ManifestTemplate {
+
+  public static final String MANIFEST_MEDIA_TYPE =
+      "application/vnd.docker.distribution.manifest.list.v2+json";
+  private static final int SCHEMA_VERSION = 2;
+
+  @Override
+  public int getSchemaVersion() {
+    return SCHEMA_VERSION;
+  }
+
+  private List<ManifestDescriptorTemplate> manifests = new ArrayList<>();
+
+  @VisibleForTesting
+  List<ManifestDescriptorTemplate> getManifests() {
+    return manifests;
+  }
+
+  public List<String> getDigestForPlatform(String architecture, String os) {
+    return manifests
+        .stream()
+        .filter(
+            manifest -> {
+              if (manifest.platform == null) {
+                return false;
+              }
+              if (!os.equals(manifest.platform.os)) {
+                return false;
+              }
+              if (!architecture.equals(manifest.platform.architecture)) {
+                return false;
+              }
+              return true;
+            })
+        .map(ManifestDescriptorTemplate::getDigest)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Template for inner JSON object representing content descriptor for a layer or container
+   * configuration.
+   *
+   * @see <a href="https://github.com/opencontainers/image-spec/blob/master/descriptor.md">OCI
+   *     Content Descriptors</a>
+   */
+  static class ManifestDescriptorTemplate implements JsonTemplate {
+
+    static class Platform implements JsonTemplate {
+      @Nullable private String architecture;
+      @Nullable private String os;
+      // ignored properties
+      @Nullable
+      @JsonProperty("os.version")
+      private String osVersion;
+
+      @Nullable
+      @JsonProperty("os.features")
+      private List<String> osFeatures;
+
+      @Nullable private String variant;
+      @Nullable private List<String> features;
+
+      /** Required by Jackson-JSON */
+      private Platform() {}
+
+      @Nullable
+      String getArchitecture() {
+        return architecture;
+      }
+
+      @Nullable
+      public String getOs() {
+        return os;
+      }
+    }
+
+    @Nullable private String mediaType;
+    @Nullable private String digest;
+    private long size;
+    @Nullable private Platform platform;
+
+    /** Required by Jackson-JSON */
+    public ManifestDescriptorTemplate() {}
```

Comment:
Interesting. Shouldn't this constructor be implicit and exist anyways?

Reply:
Yeah, for some reason I was getting an error with jackson, but that might be because I was using non-static inner classes. Let me try again.